irispy/iris_tools.py

```diff
@@ -212,6 +220,88 @@ def get_iris_response(pre_launch=False, response_file=None, response_version=Non
     return iris_response
 
 
+def fit_iris_xput(time_obs, time_cal_coeffs, cal_coeffs):
+    """
+    To calculate the coefficients of best-fit time function for throughput,
+    for which there are two modes:
+    1. Perform fit: supply xput and single element ``cal_coeffs``.
+    2. Apply fit: supply full ``cal_coeffs``.
+
+    Parameters
+    ----------
+    time_obs: a numpy array of floats
+    - Times of datapoints.
+
+    time_cal_coeffs: a numpy array of floats
+    - Start and end times of intervals of constant ``cal_coeffs[i]``.
+
+    cal_coeffs: a numpy array of floats
+    - Coefficients of best-fit function.
+
+    Returns
+    -------
+    Yields the values of fit for times ``time_obs``.
+
+    """
+    if isinstance(time_cal_coeffs, np.ndarray):
+        if isinstance(cal_coeffs, np.ndarray):
+            size_time_cal_coeffs = time_cal_coeffs.shape
+            size_cal_coeffs = cal_coeffs.shape
+
```

Comment:
Convert ```time_obs``` to an array of ```datetime.datetime``` objects near the start of the function.  The translations to do this are [here](https://github.com/sunpy/irispy/pull/108/files#diff-dee23d5475a6ed5e368938455c92fb2eR266).  Then later we can make the input take the form of ```datetime.datetime``` or ```astropy.time.Time``` without messing up the output.